python/cudf/cudf_pandas_tests/test_cudf_pandas_no_fallback.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from cudf.pandas import LOADED
+
+if not LOADED:
+    raise ImportError("These tests must be run with cudf.pandas loaded")
+
+import numpy as np
+import pandas as pd
+
+
+@pytest.fixture(autouse=True)
+def fail_on_fallback(monkeypatch):
+    monkeypatch.setenv("CUDF_PANDAS_FAIL_ON_FALLBACK", "True")
```

Comment:
Will this have to be unset and is the scope of the fixture guaranteed to be limited to this file when ran with pytest-xdist with `worksteal` algo?

Reply:
According to the [pytest docs](https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session), the default scope for the fixture is at the function level. So I think each test gets the environment variable set and unset on startup and teardown, respectively.

Reply:
Also, it doesn't look like [we don't run `worksteal`](https://github.com/rapidsai/cudf/blob/branch-24.10/ci/cudf_pandas_scripts/run_tests.sh) for the cudf.pandas test suite. Should we?

Reply:
Yup, can you add that as part of this pr